src/finchlite/autoschedule/compiler.py

```diff
@@ -204,11 +203,7 @@ def __call__(
                 return ntn.Assign(
                     ntn.Variable(
                         name,
-                        BufferizedNDArrayFType(
-                            NumpyBufferFType(val.dtype),
-                            val.ndim,
-                            TupleFType.from_tuple(val.shape_type),
-                        ),
+                        val.from_kwargs(val.to_kwargs()),
```

Comment:
Proper handling of the format inference is a separate larger task (to map `traits.jl`), here I only convert it to a "dict of attributes" where I can override some of them (for example promoted dtype from two inputs).